### PR TITLE
Reorganize code structure to reduce go SDK memory usage

### PR DESCRIPTION
**Pull request by yilunzhang**
### Type (put an `x` where ever applicable)
- [ ] Bug fix: Link to the issue
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: chain/blockvalidator.go [hunk 1]
```diff
@@ -29,7 +32,12 @@ const (
 	NumGenesisBlocks           = 4
 )
 
-var timestampToleranceSalt []byte = util.RandomBytes(32)
+var (
+	ErrIDRegistered           = errors.New("ID has be registered")
```
Comment (bufrr) [inline chain/blockvalidator.go:36]: typo?

File: block/block.go [hunk 2]
```diff
@@ -257,7 +257,7 @@ func (b *Block) GetInfo() ([]byte, error) {
 	info := &blockInfo{
 		Header:       unmarshaledHeader,
 		Transactions: make([]interface{}, 0),
-		Size:         b.ToMsgBlock().Size(),
+		Size:         proto.Size(b.ToMsgBlock()),
```
Comment (bufrr) [inline block/block.go:260]: why chosing `proto.Size` instead? And they both return the same value in this case right?

File: chain/blockvalidator.go (hunk 1 above)
Comment (yilunzhang) [reply chain/blockvalidator.go]: You are right, fixed

File: block/block.go (hunk 2 above)
Comment (yilunzhang) [reply block/block.go:260]: Google's pb only has this API, it returns the same result

**Review by bufrr — APPROVED** [review]

**Pull request merged**